File: app/src/main/java/com/fonfon/noloss/db/DeviceDB.java
package com.fonfon.noloss.db;

import com.fonfon.noloss.lib.Device;

import static com.fonfon.noloss.lib.Device.ZERO_GEOHASH;

public class DeviceDB {

  private Long _id;
  private String address;
  private String name;
  private String geoHash;
  private String image;

  public DeviceDB() {
  }

  public DeviceDB(String address, String name, String defaultImage) {
    this._id = Device.doHash(address);
    this.address = address;
    this.name = name;
    geoHash = ZERO_GEOHASH;
    this.image = defaultImage;
  }

  public DeviceDB(Device device) {
    this._id = device.get_id();
    this.address = device.getAddress();
    this.name = device.getName();
    this.geoHash = device.getGeoHash();
    this.image = device.getImage();
  }

  public Long get_id() {
    return _id;
  }

  public String getAddress() {
    return address;
  }

  public void setAddress(String address) {
    this.address = address;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getGeoHash() {
    return geoHash;
  }

  public void setGeoHash(String geoHash) {
    this.geoHash = geoHash;
  }

  public String getImage() {
    return image;
  }

  public void setImage(String image) {
    this.image = image;
  }
}

File: app/src/main/java/com/fonfon/noloss/db/DeviceDB.kt
package com.fonfon.noloss.db

import android.location.Location

import com.fonfon.geohash.GeoHash
import com.fonfon.noloss.lib.Device

import com.fonfon.noloss.lib.Device.ZERO_GEOHASH

class DeviceDB {

  var _id: Long? = null
  var address: String? = null
  var name: String? = null
  var geoHash: String? = null
  var image: String? = null

  constructor() {}

  constructor(address: String, name: String, defaultImage: String, location: Location?) {
    this._id = Device.doHash(address)
    this.address = address
    this.name = name
    geoHash = if (location == null) ZERO_GEOHASH else GeoHash.fromLocation(location, GeoHash.MAX_CHARACTER_PRECISION).toString()
    this.image = defaultImage
  }

  constructor(device: Device) {
    this._id = device._id
    this.address = device.address
    this.name = device.name
    this.geoHash = device.geoHash
    this.image = device.image
  }
}
